packages/transaction-controller/src/helpers/PendingTransactionTracker.ts

```diff
@@ -126,23 +125,22 @@ export class PendingTransactionTracker {
     this.#isResubmitEnabled = isResubmitEnabled ?? true;
     this.#listener = this.#onLatestBlock.bind(this);
     this.#nonceTracker = nonceTracker;
-    this.#onStateChange = onStateChange;
     this.#publishTransaction = publishTransaction;
     this.#running = false;
     this.#beforePublish = hooks?.beforePublish ?? (() => true);
     this.#beforeCheckPendingTransaction =
       hooks?.beforeCheckPendingTransaction ?? (() => true);
+  }
 
-    this.#onStateChange(() => {
-      const pendingTransactions = this.#getPendingTransactions();
+  onStateChange = () => {
```

Comment:
Since this is now invoked externally, this helper conceptually needs no knowledge of why it was invoked.

To decouple it a little and add clarity, could we call this something like `startIfPendingTransactions`?